python/dolfinx_contact/unbiased/nitsche_unbiased.py

```diff
@@ -0,0 +1,265 @@
+        J += - ufl.inner(sigma(du) * n, v) * ds(dirichlet_value_1)\
+            - theta * ufl.inner(sigma(v) * n, du) * \
+            ds(dirichlet_value_1) + gamma / h * ufl.inner(du, v) * ds(dirichlet_value_1)
+    else:
+        raise RuntimeError("Strong Dirichlet bc's are not implemented in custom assemblers yet.")
+
+    # Custom assembly
+    _log.set_log_level(_log.LogLevel.OFF)
+    # create contact class
+    contact = dolfinx_contact.cpp.Contact(facet_marker, surface_value_0, surface_value_1, V._cpp_object)
+    contact.set_quadrature_degree(quadrature_degree)
+    contact.create_distance_map(0)
+    contact.create_distance_map(1)
+    # pack constants
+    consts = np.array([gamma, theta])
+
+    # Pack material parameters mu and lambda on each contact surface
+    V2 = _fem.FunctionSpace(mesh, ("DG", 0))
+    lmbda2 = _fem.Function(V2)
+    lmbda2.interpolate(lambda x: np.full((1, x.shape[1]), lmbda))
+    mu2 = _fem.Function(V2)
+    mu2.interpolate(lambda x: np.full((1, x.shape[1]), mu))
+    facets_0 = facet_marker.indices[facet_marker.values == surface_value_0]
+    facets_1 = facet_marker.indices[facet_marker.values == surface_value_1]
+
+    integral = _fem.IntegralType.exterior_facet
+    entities_0 = dolfinx_cuas.compute_active_entities(mesh, facets_0, integral)
+    material_0 = dolfinx_cuas.pack_coefficients([mu2, lmbda2], entities_0)
+
+    entities_1 = dolfinx_cuas.compute_active_entities(mesh, facets_1, integral)
+    material_1 = dolfinx_cuas.pack_coefficients([mu2, lmbda2], entities_1)
+
+    # Pack circumradius on each surface
+    h_0 = dolfinx_contact.pack_circumradius_facet(mesh, facets_0)
+    h_1 = dolfinx_contact.pack_circumradius_facet(mesh, facets_1)
+
+    # Pack gap, normals and test functions on each surface
+    gap_0 = contact.pack_gap(0)
+    n_0 = contact.pack_ny(0, gap_0)
+    test_fn_0 = contact.pack_test_functions(0, gap_0)
+    gap_1 = contact.pack_gap(1)
+    n_1 = contact.pack_ny(1, gap_1)
+    test_fn_1 = contact.pack_test_functions(1, gap_1)
+
+    # Concatenate all coeffs
+    coeff_0 = np.hstack([material_0, h_0, gap_0, n_0, test_fn_0])
+    coeff_1 = np.hstack([material_1, h_1, gap_1, n_1, test_fn_1])
+
+    # Assemble jacobian
+    J_cuas = _fem.Form(J)
+    kernel_jac = contact.generate_kernel(kt.Jac)
+
+    def create_A():
+        return contact.create_matrix(J_cuas._cpp_object)
+
+    def A(x, A):
+        u.vector[:] = x.array
+        u_opp_0 = contact.pack_u_contact(0, u._cpp_object, gap_0)
+        u_opp_1 = contact.pack_u_contact(1, u._cpp_object, gap_1)
+        u_0 = dolfinx_cuas.pack_coefficients([u], entities_0)
+        u_1 = dolfinx_cuas.pack_coefficients([u], entities_1)
+        c_0 = np.hstack([coeff_0, u_0, u_opp_0])
+        c_1 = np.hstack([coeff_1, u_1, u_opp_1])
+        contact.assemble_matrix(A, [], 0, kernel_jac, c_0, consts)
+        # To create onesided contact assembly remove comment
```

Comment:
I guess these are strictly not true (as we also need to modify the kernel). So I guess we can just remove the comments here and below